### PR TITLE
Improve waiting for task end

### PR DESCRIPTION
**Pull request by index-git**
Part of issue #

- [x] Tests
- ~~Layman Test Client~~
- ~~Changlelog~~
- ~~Documentation~~

Optionally
- Add dependency to doc/dependencies.md

### CODE REVIEW

File: test_tools/util.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from requests.exceptions import ConnectionError
```
Comment (jirik) [inline test_tools/util.py:5]: Try to find some downloadable big file, e.g. in NaturalEarth, and use it instead of africa_surface_lithology.zip

**Review by jirik — CHANGES_REQUESTED** [review]

File: src/layman/celery.py [hunk 2]
```diff
@@ -204,18 +200,28 @@ def set_publication_chain_info(workspace, publication_type, publication_name, ta
...
 
-def abort_chain(chain_info):
+def wait_for_abort(workspace, publication_type, publication_name):
+    round = 0
+    max_rounds = 10
+    while round <= max_rounds:
+        chain_info = get_publication_chain_info_dict(workspace, publication_type, publication_name)
+        if chain_info['finished']:
+            break
+        time.sleep(0.3)
```
Comment (jirik) [inline src/layman/celery.py:210]: 0.3 -> 0.5

File: src/layman/celery.py [hunk 3]
```diff
@@ -204,18 +200,28 @@ def set_publication_chain_info(workspace, publication_type, publication_name, ta
     rds.hset(key, hash, val)
 
 
-def abort_chain(chain_info):
+def wait_for_abort(workspace, publication_type, publication_name):
+    round = 0
+    max_rounds = 10
```
Comment (jirik) [inline src/layman/celery.py:205]: 10 -> 20

File: src/layman/celery.py [hunk 4]
```diff
@@ -49,14 +49,6 @@ def task_postrun(workspace, publication_type, publication_name, task_id, task_na
             last_task_id = chain_info['last']
```
Comment (jirik) [inline src/layman/celery.py:49]: Move finish_publication_chain (line 38) after update_related_publications_after_change

File: src/layman/celery.py [hunk 5]
```diff
@@ -49,14 +49,6 @@ def task_postrun(workspace, publication_type, publication_name, task_id, task_na
...
             finish_publication_chain(last_task_id, task_state)
```
Comment (jirik) [inline src/layman/celery.py:50]: Move this line after next one

File: src/layman/celery.py [hunk 6]
```diff
@@ -113,10 +112,7 @@ def finish_publication_chain(last_task_id_in_chain, state):
         return
     workspace, publication_type, publication_name = _hash_to_publication(publ_hash)
 
-    chain_info = get_publication_chain_info_dict(workspace, publication_type, publication_name)
-    chain_info['finished'] = True
-    chain_info['state'] = state
-    set_publication_chain_info_dict(workspace, publication_type, publication_name, chain_info)
+    set_publication_chain_finished(workspace, publication_type, publication_name, state)
```
Comment (jirik) [inline src/layman/celery.py:115]: Move this line after next one

**Review by jirik — CHANGES_REQUESTED** [review]

**Review by jirik — APPROVED** [review]

File: src/layman/celery.py [hunk 7]
```diff
@@ -35,28 +35,20 @@ def task_postrun(workspace, publication_type, publication_name, task_id, task_na
     key = LAST_TASK_ID_IN_CHAIN_TO_PUBLICATION
     hash = task_id
     if rds.hexists(key, hash):
-        finish_publication_chain(task_id, task_state)
         next_task = pop_step_to_run_after_chain(workspace, publication_type, publication_name)
+        update_related_publications_after_change(workspace, publication_type, publication_name)
+        finish_publication_chain(task_id, task_state)
```
Comment (index-git) [inline src/layman/celery.py:40]: finish_publication_chain open the lock, and the "method" on line 45 expects the publication to be unlocked, so this order is needed.

**Review by jirik — APPROVED** [review]

**Pull request merged**